appinventor/appengine/src/com/google/appinventor/client/boxes/PropertiesBox.java

```diff
@@ -38,5 +56,93 @@ private PropertiesBox() {
+    }
+    if (components.size() == 1) {
+      selectedProperties = components.get(0).getProperties();
+    } else {
+      EditableProperties newProperties = new EditableProperties(true);
+      Map<String, EditableProperty> propertyMaps = new HashMap<>();
+      boolean first = true;
+      for (MockComponent component : components) {
+        Set<String> properties = new HashSet<>();
+        for (EditableProperty property : component.getProperties()) {
+          String propertyName = property.getName();
+          // Ignore UUID and NAME properties (can't be edited and always unique)
+          if ("Uuid".equals(propertyName) || "Name".equals(propertyName)) {
+            continue;
+          }
+          if (first) {
+            propertyMaps.put(propertyName + ":" + property.getType(), property);
+          } else {
+            properties.add(propertyName + ":" + property.getType());
+          }
+        }
+        if (properties.size() > 0) {
+          propertyMaps.keySet().retainAll(properties);
+        }
+        first = false;
+      }
+      for (EditableProperty property : propertyMaps.values()) {
+        String name = property.getName();
+        newProperties.addProperty(
+            name,
+            property.getDefaultValue(),
+            property.getCaption(),
+            PropertiesUtil.createPropertyEditor(property.getEditorType(),
+                property.getDefaultValue(), formEditor , property.getEditorArgs()),
+            property.getType(),
+            property.getEditorType(),
+            property.getEditorArgs()
+        );
+
+        // Determine if all components have the same value and apply it
+        String sharedValue = components.get(0).getPropertyValue(name);
+        boolean collision = false;
+        for (MockComponent component : components) {
+          String propValue = component.getPropertyValue(name);
+          if (!sharedValue.equals(propValue)) {
+            sharedValue = "";
+            collision = true;
+            break;
+          }
+        }
+        newProperties.getProperty(name).getEditor().setMultipleValues(collision);
+        newProperties.getProperty(name).getEditor().setMultiselectMode(true);
+        newProperties.getProperty(name).setValue(sharedValue);
+      }
+      selectedProperties = newProperties;
+    }
+    if (selected) {
+      selectedProperties.addPropertyChangeListener(formEditor);
+    }
+    designProperties.setProperties(selectedProperties);
+    if (components.size() > 1) {
+      // TODO: Localize
+      designProperties.setPropertiesCaption(components.size() + " components selected");
+    } else {
+      // need to update the caption after the setProperties call, since
+      // setProperties clears the caption!
+      designProperties.setPropertiesCaption(components.get(0).getName() + " (" +
+                                               components.get(0).getType() + ")");
+    }
+    setContent(designProperties);
```

Comment:
Is it necessary to call setContent every time? I feel like this could be done once in the constructor seeing as designProperties is `final`.